apps/frontend/src/libs/components/checkbox/checkbox.tsx

```diff
@@ -1,73 +1,131 @@
+import { ZERO_INDEX } from "~/libs/constants/constants.js";
+import { getValidClassNames } from "~/libs/helpers/helpers.js";
 import { useCallback, useFormController } from "~/libs/hooks/hooks.js";
 import {
-	type Control,
-	type FieldPath,
 	type FieldValues,
+	type FormFieldProperties,
 	type InputOption,
 } from "~/libs/types/types.js";
 
 import styles from "./styles.module.css";
 
 type Properties<T extends FieldValues> = {
-	control: Control<T, null>;
+	hasVisuallyHiddenLabel?: boolean;
 	label: string;
-	name: FieldPath<T>;
+	onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
 	options: InputOption[];
-};
+	variant: "general" | "rounded";
+} & FormFieldProperties<T>;
 
 const Checkbox = <T extends FieldValues>({
 	control,
+	hasVisuallyHiddenLabel = false,
 	label,
 	name,
+	onChange,
 	options,
+	variant,
 }: Properties<T>): JSX.Element => {
 	const { field } = useFormController({ control, name });
-	const { onChange, value } = field;
+	const { onChange: onFieldChange, value } = field;
+	const isGeneralCheckbox = variant === "general";
```

Comment:
So its better to switch it as `isRounded` variant?

Reply:
Or since this enum prop is also affect the component layout, is it more suitable to rename this prop as `type` with `"category" | "question"` as its type?

Reply:
If it is only responsible for making it rounded, then use `isRounded` prop